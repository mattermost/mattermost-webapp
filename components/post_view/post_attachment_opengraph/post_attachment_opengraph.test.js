// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PostAttachmentOpenGraph from './post_attachment_opengraph.jsx';

describe('components/post_view/PostAttachmentOpenGraph', () => {
    const post = {
        id: 'post_id_1',
        root_id: 'root_id',
        channel_id: 'channel_id',
        create_at: 1,
        message: 'https://mattermost.com',
        metadata: {
            images: {
                'http://mattermost.com/OpenGraphImage.jpg': {
                    height: 100,
                    width: 100,
                },
            },
        },
    };

    const baseProps = {
        post,
        link: 'http://mattermost.com',
        previewEnabled: true,
        isEmbedVisible: true,
        enableLinkPreviews: true,
        hasImageProxy: true,
        currentUser: {
            id: '1234',
        },
        openGraphData: {
            description: 'description',
            images: [{
                height: 100,
                secure_url: '',
                url: 'http://mattermost.com/OpenGraphImage.jpg',
                width: 100,
            }],
            site_name: 'Mattermost',
            title: 'Mattermost Private Cloud Messaging',
        },
        toggleEmbedVisibility: jest.fn(),
        actions: {
            editPost: jest.fn(),
            getOpenGraphMetadata: jest.fn(),
        },
    };

    test('Match snapshot for no openGraphData', () => {
        const props = {
            ...baseProps,
            openGraphData: null,
        };

        const wrapper = shallow(
            <PostAttachmentOpenGraph {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Match snapshot for small image openGraphData', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Match snapshot for large image openGraphData', () => {
        const props = {
            ...baseProps,
            openGraphData: {
                ...baseProps.openGraphData,
                images: [{
                    height: 180,
                    secure_url: '',
                    url: 'http://mattermost.com/OpenGraphImage.jpg',
                    width: 400,
                }],
            },
            post: {
                ...post,
                metadata: {
                    ...post.metadata,
                    images: {
                        'http://mattermost.com/OpenGraphImage.jpg': {
                            height: 180,
                            width: 400,
                        },
                    },
                },
            },
        };

        const wrapper = shallow(
            <PostAttachmentOpenGraph {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Match snapshot for with remove preview', () => {
        const props = {
            ...baseProps,
            post: {
                ...post,
                user_id: '1234',
            },
        };

        const wrapper = shallow(
            <PostAttachmentOpenGraph {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
