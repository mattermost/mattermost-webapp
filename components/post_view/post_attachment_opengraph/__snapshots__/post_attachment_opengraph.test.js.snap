// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/PostAttachmentOpenGraph Match snapshot for large image openGraphData 1`] = `
<div
  className="attachment attachment--opengraph"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container attachment__container--opengraph"
    >
      <div
        className="attachment__body__wrap attachment__body__wrap--opengraph"
      >
        <span
          className="sitename"
        >
          Mattermost
        </span>
        <h1
          className="attachment__title attachment__title--opengraph"
        >
          <a
            className="attachment__title-link attachment__title-link--opengraph"
            href="http://mattermost.com"
            rel="noopener noreferrer"
            target="_blank"
            title="Mattermost Private Cloud Messaging"
          >
            Mattermost Private Cloud Messaging
          </a>
        </h1>
        <div
          className="attachment__body attachment__body--opengraph"
        >
          <div>
            <div>
              description
               
            </div>
          </div>
        </div>
      </div>
      <div
        className="attachment__image__container--opengraph"
      >
        <SizeAwareImage
          className="attachment__image attachment__image--opengraph"
          dimensions={null}
          onImageLoaded={[Function]}
          src="/api/v4/image?url=http%3A%2F%2Fmattermost.com%2FOpenGraphImage.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/post_view/PostAttachmentOpenGraph Match snapshot for no openGraphData 1`] = `""`;

exports[`components/post_view/PostAttachmentOpenGraph Match snapshot for small image openGraphData 1`] = `
<div
  className="attachment attachment--opengraph"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container attachment__container--opengraph"
    >
      <div
        className="attachment__body__wrap attachment__body__wrap--opengraph"
      >
        <span
          className="sitename"
        >
          Mattermost
        </span>
        <h1
          className="attachment__title attachment__title--opengraph"
        >
          <a
            className="attachment__title-link attachment__title-link--opengraph"
            href="http://mattermost.com"
            rel="noopener noreferrer"
            target="_blank"
            title="Mattermost Private Cloud Messaging"
          >
            Mattermost Private Cloud Messaging
          </a>
        </h1>
        <div
          className="attachment__body attachment__body--opengraph"
        >
          <div>
            <div>
              description
               
            </div>
          </div>
        </div>
      </div>
      <div
        className="attachment__image__container--opengraph"
      >
        <SizeAwareImage
          className="attachment__image attachment__image--opengraph"
          dimensions={null}
          onImageLoaded={[Function]}
          src="/api/v4/image?url=http%3A%2F%2Fmattermost.com%2FOpenGraphImage.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/post_view/PostAttachmentOpenGraph Match snapshot for with remove preview 1`] = `
<div
  className="attachment attachment--opengraph"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container attachment__container--opengraph"
    >
      <div
        className="attachment__body__wrap attachment__body__wrap--opengraph"
      >
        <span
          className="sitename"
        >
          Mattermost
        </span>
        <button
          aria-label="Close"
          className="btn-close"
          id="removePreviewButton"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
        <h1
          className="attachment__title attachment__title--opengraph"
        >
          <a
            className="attachment__title-link attachment__title-link--opengraph"
            href="http://mattermost.com"
            rel="noopener noreferrer"
            target="_blank"
            title="Mattermost Private Cloud Messaging"
          >
            Mattermost Private Cloud Messaging
          </a>
        </h1>
        <div
          className="attachment__body attachment__body--opengraph"
        >
          <div>
            <div>
              description
               
            </div>
          </div>
        </div>
      </div>
      <div
        className="attachment__image__container--opengraph"
      >
        <SizeAwareImage
          className="attachment__image attachment__image--opengraph"
          dimensions={null}
          onImageLoaded={[Function]}
          src="/api/v4/image?url=http%3A%2F%2Fmattermost.com%2FOpenGraphImage.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
